design/fips-compliance.md

```diff
@@ -0,0 +1,259 @@
+   
+| **Scenario**            | **Given**                                                    | **When**                                                 | **Then**                                                                                               |
+|-------------------------|--------------------------------------------------------------|----------------------------------------------------------|--------------------------------------------------------------------------------------------------------|
+| Authentication succeeds | A Running Conjur cluster with a configured K8s Authenticator | I run the authenticator client with a valid k8s host     | An access token is provided to the application container and it can retrieve a secret with it          |
+| Authentication fails    | A Running Conjur cluster with a configured K8s Authenticator | I run the authenticator client with a non-valid k8s host | An access token is not provided to the application container and the log shows `CAKC015E Login failed` |
+
+## Docs
+
+We should document any change that will affect the customer (e.g if we release 
+both FIPS & Non-FIPS versions).
+
+If no customer-facing changes are introduced, we will not need to add documentation.
+
+## Version update
+
+We should update the version of:
+  - `conjur-authn-k8s-client`
+  - `seed-fetcher`
+  
+We don't need to update the version of the `secretless-broker` and the
+`secrets-provider-for-k8s` as they only consume libraries of the `conjur-authn-k8s-client`
+and will not be affected by the base image change. 
+
+## Security
+
+It's important to note that this solution is secure. Google is maintaining this 
+fork of Go that uses BoringCrypto and they [intend to maintain in this branch 
+the latest release plus BoringCrypto patches](https://go.googlesource.com/go/+/refs/heads/dev.boringcrypto/README.boringcrypto.md).
+It's also worth to link to [FIPS certification of the crypto library of dev.boringcrypto](https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3318.pdf).
+
+Furthermore, the DoD of this effort is to be able to tell customers that we are using only 
+FIPS compliant crypto libraries and not that our product is FIPS compliant by itself. 
+Our solution meets that requirement so it meets the security needs.
+
+## DoD
+
+- [ ] Solution design is approved by the following personas
+  - [ ] Team Leader: Dekel Asaf (@Tovli)
+  - [ ] Product Owner: Hila Gross (@hilagross)
+  - [ ] Repository Owner: Geri Jennings (@izgeri)
+  - [ ] QA Architect: Elad Kugman (@eladkug)
+  - [ ] System Architect: Ofira Burstein (@oburstein-hub)
+  - [ ] Security Architect: Shahar Glazner: (@shaharglazner)
+- [ ] `conjur-authn-k8s-client` is FIPS compliant
+- [ ] Tests are implemented according to Test Plan and are passing
+- [ ] Required documentation changes are implemented
+- [ ] Versions are bumped in all relevant projects
+
```

Comment:
From meeting:
Add `Open Issues` section

1. Release two separate images?
2. Deal with integration tests or add to existing infrastructure (add your opinion)?

Reply:
present mitigation plan for the open issues